tests/unit/model/test_can_approve.py

```diff
@@ -0,0 +1,238 @@
+# you may not use this file except in compliance with the License. You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed under the License is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+from __future__ import unicode_literals
+from copy import deepcopy
+from functools import partial
+
+import couch
+import pytest
+from mock import patch
+from tornado.ioloop import IOLoop
+from tornado.httpclient import HTTPError
+
+from perch import Organisation, Service, Repository, UserOrganisation, User
+from ..util import make_future
+
+USER = {
+    '_id': 'user0',
+    'type': 'user',
+    'email': 'user0@mail.test',
+    'password': User.hash_password('password0'),
+    'verified': True,
+    'state': 'approved',
+    'has_agreed_to_terms': True,
+    'organisations': {
+        'global': {
+            'state': 'approved',
+            'role': 'user'
+            }
+    }
+}
+
+global_joins = [
+    ({'global': {
+        'state': 'approved',
+        'role': 'administrator'
+    }}, True),
+    ({'global': {
+        'state': 'approved',
+        'role': 'user'
+    }}, False)
+]
+
+user_joins = [
+    ({
+        'state': 'pending',
+        'role': 'user'
+    }, False),
+    ({
+        'state': 'approved',
+        'role': 'user'
+    }, False),
+    ({
+        'state': 'rejected',
+        'role': 'user'
+    }, False),
+    ({
+        'state': 'deactivated',
+        'role': 'user'
+    }, False),
+]
+
+admin_joins = [
+    ({
+        'state': 'pending',
+        'role': 'administrator'
+    }, False),
+    ({
+        'state': 'approved',
+        'role': 'administrator'
+    }, True),
+    ({
+        'state': 'rejected',
+        'role': 'administrator'
+    }, False),
+    ({
+        'state': 'deactivated',
+        'role': 'administrator'
+    }, False)
+]
+
+class TestOrganisation():
+    organisation = Organisation(id='org0')
+
+    @pytest.mark.parametrize("org_info,expected", global_joins)
+    def test_can_approve(self, org_info, expected):
+        u = deepcopy(USER)
+        u['organisations'] = org_info
+        user = User(**u)
+        func = partial(self.organisation.can_approve, user)
+        result = IOLoop.instance().run_sync(func)
+        assert result == expected
+
+class TestService():
+
+    @pytest.mark.parametrize("org_info,expected", global_joins)
+    def test_can_approve_external(self, org_info, expected):
+        service = Service(id='serv0', service_type="external")
+        u = deepcopy(USER)
+        u['organisations'] = org_info
+        user = User(**u)
+        func = partial(service.can_approve, user)
+        result = IOLoop.instance().run_sync(func)
+        assert result is True
```

Comment:
External services should always be approvable, so it will always be True. The expected value in the test data is for non-external services.